exonum-java-binding/core/src/main/java/com/exonum/binding/core/storage/database/MemoryDb.java

```diff
@@ -70,12 +71,20 @@ public Fork createFork(Cleaner cleaner) {
   }
 
   /**
-   * Applies the changes from the given fork to the database state.
+   * Applies the changes from the given fork to the database state. MemoryDb can only
+   * merge forks that {@linkplain #createFork(Cleaner) it created itself}.
+   *
+   * <p>Once this method completes, any indexes created with the fork and the fork itself
```

Comment:
can we have a test that calls `merge` (or `fork.intoPatch()`) twice?

Reply:
Added. Ideally it must throw IllegalArgument, but IllegalState in the case of rarely used method will do.

Reply:
It is not possible to check if a Fork is valid without adding an extra package-private method.